galsim/chromatic.py

```diff
@@ -1359,3 +1770,76 @@ def withScaledFlux(self, flux_ratio):
             return ChromaticAutoCorrelation( self.obj * math.sqrt(flux_ratio), **self.kwargs )
         else:
             return ChromaticObject(self).withScaledFlux(flux_ratio)
+
+class ChromaticOpticalPSF(ChromaticObject):
+    """A subclass of ChromaticObject meant to represent chromatic optical PSFs.
+
+    Chromaticity plays two roles in optical PSFs. First, it determines the diffraction limit, via
+    the wavelength/diameter factor.  Second, aberrations such as defocus, coma, etc. are typically
+    defined in physical distances, but their impact on the PSF depends on their size in units of
+    wavelength.  Other aspects of the optical PSF do not require explicit specification of their
+    chromaticity, e.g., once the obscuration and struts are specified in units of the aperture
+    diameter, their chromatic dependence gets taken care of automatically.
+
+    When using interpolation to speed up image rendering (see ChromaticObject.setupInterpolation()
+    method for details), including ~10-15 samples across any given bandpass should be sufficient.
+    For moderate accuracy, 5 is often sufficient.  Likewise, for high accuracy, `oversample_fac`
+    should be in the range 1.5-2, whereas for moderate accuracy, 1 is often sufficient.  All of
+    these statements assume that aberrations are not very large (typically <~0.25 waves, which is
+    commonly satisfied by space telescopes); if they are larger than that, then more stringent
+    settings are required.
+
+    Also, note that in order to render an image of a real physical object, the ChromaticOpticalPSF
+    should be convolved with some object that has an SED.  Drawing without doing so corresponds to
+    using a flat SED.
+
+    @param   diam          Telescope diameter in meters.
+    @param   aberrations   An array of aberrations, in nanometers.  The size and format of this
+                           array is described in the OpticalPSF docstring.
```

Comment:
I agree that it would be good to document this better in OpticalPSF and ChromaticOpticalPSF.

I don't want to break old functionality, but I think your compromise is okay.  For OpticalPSF, we can allow either ```lam_over_diam``` or separate diameter and wavelength specification.  For ChromaticOpticalPSF we require both diameter and wavelength, or ```lam_over_diam``` and wavelength, and tell the user to enter aberrations in units of waves at the specified wavelength.   Will do this in a bit.